js/modals/question-modal.js

```diff
@@ -101,78 +94,67 @@ $(function() {
-      var io2 = '<button type="button" class="btn btn-default io" data-io="2"><span>2</span></button>';
-      var io3 = '<button type="button" class="btn btn-default io" data-io="3"><span>3</span></button>';
-      var $inputTextarea = $('<textarea class="form-control" id="input-textarea" rows="3" placeholder="Question Input 1"></textarea>');
-      var $outputInput = $('<input type="text" id="output-input" class="form-control" placeholder="Expected Output 1">');
-
-      $io
-          .append($removeIo)
-          .append($addIo)
-          .append(io1)
-          .append(io2)
-          .append(io3)
-          .append($inputTextarea)
-          .append($outputInput);
+
+      $('.question-type-area').attr('class', 'question-type-area coding');
+
+      var $io = $templates.find('.js-modal-question-code .js-io')
+          .clone()
+          .children();
 
       return $editor.add($io);
    }
 
    function ioButtonClick(button) {
       if (!button.hasClass('active')) {
-         $('fieldset.input-output .io.active').data('input', $('fieldset.input-output textarea').val());
-         $('fieldset.input-output .io.active').data('output', $('fieldset.input-output input').val());
-         $('fieldset.input-output textarea').val(button.data('input'));
-         $('fieldset.input-output input').val(button.data('output'));
-         $('fieldset.input-output textarea').attr('placeholder', 'Question Input ' + button.data('io'));
-         $('fieldset.input-output input').attr('placeholder', 'Expected Output ' + button.data('io'));
-         $('.io.active>span').text($('.io.active').data('io'));
-         $('.io.active').removeClass('active');
+         $('.modal fieldset.input-output .io.active').data('input', $('.modal fieldset.input-output textarea').val());
+         $('.modal fieldset.input-output .io.active').data('output', $('.modal fieldset.input-output input').val());
+         $('.modal fieldset.input-output textarea').val(button.data('input'));
+         $('.modal fieldset.input-output input').val(button.data('output'));
+         $('.modal fieldset.input-output textarea').attr('placeholder', 'Question Input ' + button.data('io'));
+         $('.modal fieldset.input-output input').attr('placeholder', 'Expected Output ' + button.data('io'));
+         $('.modal .io.active>span').text($('.modal .io.active').data('io'));
+         $('.modal .io.active').removeClass('active');
 
          button
              .addClass('active')
              .find('span')
              .text('IO ' + button.data('io'));
 
-         $('fieldset.input-output textarea').focus();
+         $('.modal fieldset.input-output textarea').focus();
       }
    }
 
    function bindCodeAreaEvents() {
-      $('.modal').on('click', '.remove-io', function() {
-         if ($('.io').length > 3) {
-            $('.io').last().remove();
+      $('#q-and-a-plugin').on('click', '.modal .remove-io', function() {
```

Comment:
If you need to use the same element over and over again, either using chaining or save it into a variable.

```js
var $el = $('#q-and-a-plugin');

$el
  .on(....)
  .on(....)
  .on(....);
```
or

```js
var $el = $('#q-and-a-plugin');

$el.on(....);
$el.on(....);
$el.on(....);
```